File: BungiiAutomation/Bungii.Test.Regression.Web.Integration/Pages/Driver/Driver_DocumentationPage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace Bungii.Android.Regression.Test.Integration.Pages.Driver
{
    public class Driver_DocumentationPage
    {
        public Driver_DocumentationPage(IWebDriver webdriver)
        {
            PageFactory.InitElements(webdriver, this);
        }

        //Documentation - Header
        [FindsBy(How = How.XPath, Using = "//div[@id='tab-title']/h3")]
        public IWebElement Header_Documentation { get; set; }

        //Pickup Information - Blank field validation
        [FindsBy(How = How.Id, Using = "summary3")]
        public IWebElement Err_Documentation_AllBlank { get; set; }

        //Documentation - Driver’s License Image Upload
        [FindsBy(How = How.XPath, Using = "//div[@id='dropzone3']")]
        public IWebElement DropZone3_LicenseImage { get; set; }

        //Documentation - Driver’s License Image Upload - Remove file
        [FindsBy(How = How.XPath, Using = "//div[@id='dropzone3']/div/a[contains(text(),'Remove')]")]
        public IWebElement Link_LicenseRemoveFile { get; set; }

        //Documentation - License number
        [FindsBy(How = How.Id, Using = "DriverDocument_DocumentNo")]
        public IWebElement TextBox_LicenseNumber { get; set; }

        //Documentation - License number - Error
        [FindsBy(How = How.Id, Using = "DriverDocument_DocumentNo-error")]
        public IWebElement Err_LicenseNumber { get; set; }

        //Documentation - License expiration
        [FindsBy(How = How.Id, Using = "DriverDocument_LicenseExpiry")]
        public IWebElement TextBox_LicenseExpiry { get; set; }

        //Documentation - License expiration - Error
        [FindsBy(How = How.Id, Using = "DriverDocument_LicenseExpiry-error")]
        public IWebElement Err_LicenseExpiry { get; set; }

        //Documentation - Insurance Image Upload
        [FindsBy(How = How.XPath, Using = "//div[@id='dropzone4']")]
        public IWebElement DropZone4_InsuranceImage { get; set; }

        //Documentation - Insurance Image Upload - Remove file
        [FindsBy(How = How.XPath, Using = "//div[@id='dropzone4']/div/a[contains(text(),'Remove')]")]
        public IWebElement Link_InsuranceRemoveFile { get; set; }

        //Documentation - Insurance expiration
        [FindsBy(How = How.Id, Using = "DriverDocument_InsuranceExpiry")]
        public IWebElement TextBox_InsuranceExpiry { get; set; }

        //Documentation - Insurance expiration - Error
        [FindsBy(How = How.Id, Using = "DriverDocument_InsuranceExpiry-error")]
        public IWebElement Err_InsuranceExpiry { get; set; }

        //Documentation - Next Button
        [FindsBy(How = How.Id, Using = "btnDoc")]
        public IWebElement Button_DocNext { get; set; }
    }
}

File: Bungii.Test.Integration/src/test/java/com/bungii/web/pages/driver/Driver_DocumentationPage.java
package com.bungii.web.pages.driver;

import com.bungii.common.core.PageBase;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.How;

public class Driver_DocumentationPage extends PageBase {
    //Pickup Information - Blank field validation
    public WebElement Err_Documentation_AllBlank () { return findElement("summary3", LocatorType.Id); }

    //Documentation - driver’s License Image Upload
    public WebElement DropZone3_LicenseImage () { return findElement("//div[@id='dropzone3']", LocatorType.XPath); }

    //Documentation - driver’s License Image Upload - Remove file
    public WebElement Link_LicenseRemoveFile () { return findElement("//div[@id='dropzone3']/div/a[contains(text(),'Remove')]", LocatorType.XPath); }

    //Documentation - License number
    public WebElement TextBox_LicenseNumber () { return findElement("DriverDocument_DocumentNo", LocatorType.Id); }

    //Documentation - License number - Error
    public WebElement Err_LicenseNumber () { return findElement("DriverDocument_DocumentNo-error", LocatorType.Id); }

    //Documentation - License expiration
    public WebElement TextBox_LicenseExpiry () { return findElement("DriverDocument_LicenseExpiry", LocatorType.Id); }

    //Documentation - License expiration - Error
    public WebElement Err_LicenseExpiry () { return findElement("DriverDocument_LicenseExpiry-error", LocatorType.Id); }

    //Documentation - Insurance Image Upload
    public WebElement DropZone4_InsuranceImage () { return findElement("//div[@id='dropzone4']", LocatorType.XPath); }

    //Documentation - Insurance Image Upload - Remove file
    public WebElement Link_InsuranceRemoveFile () { return findElement("//div[@id='dropzone4']/div/a[contains(text(),'Remove')]", LocatorType.XPath); }

    //Documentation - Insurance expiration
    public WebElement TextBox_InsuranceExpiry () { return findElement("DriverDocument_InsuranceExpiry", LocatorType.Id); }

    //Documentation - Insurance expiration - Error
    public WebElement Err_InsuranceExpiry () { return findElement("DriverDocument_InsuranceExpiry-error", LocatorType.Id); }

    //Documentation - Next Button
    public WebElement Button_DocNext () { return findElement("btnDoc", LocatorType.Id); }

    public WebElement DropZoneHiddenFileTag_LicenseImage () { return findElement("//input[@class = 'dz-hidden-input'][3]", LocatorType.XPath); }

    //public WebElement DropZoneHiddenFileTag_InsuranceImagRe () { return findElement("//input[@class = 'dz-hidden-input'][last()]", LocatorType.XPath); }
    public WebElement DropZoneHiddenFileTag_InsuranceImage () { return findElement("//input[@class = 'dz-hidden-input'][3]", LocatorType.XPath); }

}
